### PR TITLE
Adding buildspec.yml file

### PR DESCRIPTION
**Pull request by xibz**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  variables:
+    GO111MODULE: "on"
+
+phases:
+  pre_build:
+    commands:
+      - go mod download
+  build:
+    commands:
+      - SKIP_INTEG_TEST=1 make test
```
Comment (sipsma) [inline buildspec.yml]: very nit: can just make `SKIP_INTEG_TEST` a variable in the `env:` section above

**Comment by sipsma** [comment]
I'm just curious, what's the context of this change? Are we going to do builds in Codebuild?

File: buildspec.yml (hunk 1 above)
Comment (xibz) [reply buildspec.yml]: I think I prefer having the environment variable defined here just so you know what command it is tied to. But I went ahead and changed it anywhoo.

Comment (sipsma) [reply buildspec.yml]: Sure, I don't have a strong feeling either way, my main question was the other I posted about more context on the purpose of this change, I just added this nit comment along the way :-)

**Comment by xibz** [comment]
@sipsma - We will be using AWS CodeBuild to build the artifacts to use in releases

**Comment by sipsma** [comment]
> @sipsma - We will be using AWS CodeBuild to build the artifacts to use in releases
Is that just for a more controlled environment than buildkite? I'm mostly wondering why Codebuild in addition to our existing build mechanisms.

**Comment by xibz** [comment]
> Is that just for a more controlled environment than buildkite? I'm mostly wondering why Codebuild in addition to our existing build mechanisms.

Yea, BuildKite could definitely do this, but I don't think we ever envisioned it doing more than CI. Using tools that are familiar to not only our team is a huge plus and is why I went with AWS CodeBuild.

**Pull request merged**